lib/fog/openstack.rb

```diff
@@ -614,6 +576,45 @@ def self.map_endpoint_type type
       end
 
     end
-  end
 
+    def self.get_version(supported_versions, uri, auth_token, connection_options = {})
+      version_cache = "#{uri}#{supported_versions}"
+      return @version[version_cache] if @version && @version[version_cache]
+      connection = Fog::Core::Connection.new("#{uri.scheme}://#{uri.host}:#{uri.port}", false, connection_options)
+      response = connection.request(
+        :expects => [200, 204, 300],
+        :headers => {'Content-Type' => 'application/json',
+                     'Accept'       => 'application/json',
+                     'X-Auth-Token' => auth_token},
+        :method  => 'GET'
+      )
+
+      body = Fog::JSON.decode(response.body)
+
+      @version                = {} unless @version
+      @version[version_cache] = extract_version_from_body(body, supported_versions)
+    end
+
+    def self.extract_version_from_body(body, supported_versions)
+      return nil if body['versions'].empty?
+      versions = get_version_values(body['versions'])
+      version = nil
+      # order is important, preffered status should be first
+      %w(CURRENT stable SUPPORTED DEPRECATED).each do |status|
+        version = versions.find { |x| x['id'].match(supported_versions) && (x['status'] == status) }
+        break if version
+      end
+
+      version
+    end
+
+    def self.get_version_values(versions)
```

Comment:
correct, no need for that anymore - in the middle of refactoring it was used two times ;)